File: VB/B143239/PropertiesUserControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors
Imports DevExpress.XtraBars

Namespace B143239
	Partial Public Class PropertiesUserControl
		Inherits XtraUserControl
		Private Sub New()
			InitializeComponent()
		End Sub

		Public Sub New(ByVal manager As BarManager)
			Me.New()
			CreateMenu(manager)
		End Sub

		Private categoriesItem As BarCheckItem
		Private alphabeticalItem As BarCheckItem
		Private Sub CreateMenu(ByVal manager As BarManager)
			manager.DockControls.Add(dockControl)
			dockControl.BeginUpdate()

			Dim bar As New Bar(manager, "propertiesControlBar")
			bar.BeginUpdate()
			bar.StandaloneBarDockControl = dockControl
			bar.DockStyle = BarDockStyle.Standalone
			bar.CanDockStyle = BarCanDockStyle.Standalone
			bar.OptionsBar.DisableCustomization = True

			categoriesItem = New BarCheckItem(manager, True)
			categoriesItem.Caption = "By category"
			AddHandler categoriesItem.CheckedChanged, AddressOf OnBarCheckItemCheckedChanged
			bar.ItemLinks.Add(categoriesItem)

			alphabeticalItem = New BarCheckItem(manager, False)
			alphabeticalItem.Caption = "Alphabetically"
			AddHandler alphabeticalItem.CheckedChanged, AddressOf OnBarCheckItemCheckedChanged
			bar.ItemLinks.Add(alphabeticalItem)

			bar.EndUpdate()
			dockControl.EndUpdate()
		End Sub

		Private processingCheckedChanged As Boolean
		Private Sub OnBarCheckItemCheckedChanged(ByVal sender As Object, ByVal e As ItemClickEventArgs)
			If processingCheckedChanged Then
				Return
			End If
			processingCheckedChanged = True
			If sender Is categoriesItem Then
				alphabeticalItem.Checked = False
				propertyView.OptionsView.ShowRootCategories = True
			ElseIf sender Is alphabeticalItem Then
				categoriesItem.Checked = False
				propertyView.OptionsView.ShowRootCategories = False
			End If
			processingCheckedChanged = False
		End Sub

		Public Sub SetSelectedObject(ByVal selectedObject As Object)
			propertyView.SelectedObject = selectedObject
		End Sub
	End Class
End Namespace

File: CS/B143239/PropertiesUserControl.cs
using DevExpress.XtraEditors;
using DevExpress.XtraBars;

namespace B143239 {
    public partial class PropertiesUserControl :XtraUserControl {
        private PropertiesUserControl() {
            InitializeComponent();
        }

        public PropertiesUserControl(BarManager manager) : this() {
            CreateMenu(manager);
        }

        private BarCheckItem categoriesItem;
        private BarCheckItem alphabeticalItem;
        private void CreateMenu(BarManager manager) {
            manager.DockControls.Add(dockControl);
            dockControl.BeginUpdate();

            Bar bar = new Bar(manager, "propertiesControlBar");
            bar.BeginUpdate();
            bar.StandaloneBarDockControl = dockControl;
            bar.DockStyle = BarDockStyle.Standalone;
            bar.CanDockStyle = BarCanDockStyle.Standalone;
            bar.OptionsBar.DisableCustomization = true;

            categoriesItem = new BarCheckItem(manager, true);
            categoriesItem.Caption = "By category";
            categoriesItem.CheckedChanged += new ItemClickEventHandler(OnBarCheckItemCheckedChanged);
            bar.ItemLinks.Add(categoriesItem);

            alphabeticalItem = new BarCheckItem(manager, false);
            alphabeticalItem.Caption = "Alphabetically";
            alphabeticalItem.CheckedChanged += new ItemClickEventHandler(OnBarCheckItemCheckedChanged);
            bar.ItemLinks.Add(alphabeticalItem);

            bar.EndUpdate();
            dockControl.EndUpdate();
        }

        private bool processingCheckedChanged;
        private void OnBarCheckItemCheckedChanged(object sender, ItemClickEventArgs e) {
            if (processingCheckedChanged) return;
            processingCheckedChanged = true;
            if (sender == categoriesItem) {
                alphabeticalItem.Checked = false;
                propertyView.OptionsView.ShowRootCategories = true;
            } else if (sender == alphabeticalItem) {
                categoriesItem.Checked = false;
                propertyView.OptionsView.ShowRootCategories = false;
            }
            processingCheckedChanged = false;
        }

        public void SetSelectedObject(object selectedObject) {
            propertyView.SelectedObject = selectedObject;
        }
    }
}
